### PR TITLE
catch errors while getting args in capture tree

### PR DESCRIPTION
**Pull request by patricklx**
workaround for
https://github.com/glimmerjs/glimmer-vm/pull/1447

### CODE REVIEW

File: ember_debug/libs/render-tree.js [hunk 1]
```diff
@@ -67,10 +67,30 @@ class InElementSupportProvider {
...
       const [id, state] = args;
       const node = this.nodeFor(state);
-      self.setupNodeRemotes(node, id, capture);
+      let capture;
+      try {
+        capture = captureNode.call(this, ...args);
+        self.setupNodeRemotes(node, id, capture);
+      } catch (e) {
+        this.nodeFor = () => {
+          this.nodeFor = nodeFor;
```
Comment (RobbieTheWagner) [inline ember_debug/libs/render-tree.js:79]: Can you please explain this? I'm confused why we would set this to a function that then sets itself to something else immediately.

Comment (patricklx) [reply ember_debug/libs/render-tree.js:79]: CaptureNode will call this function also for children nodes, so need to reset to original immediately

**Comment by RobbieTheWagner** [comment]
@patricklx should this be fixed in glimmer, instead of here?

**Comment by patricklx** [comment]
Maybe, if it can be backported. 
@NullVoxPopuli ? What do you think.

**Review by lifeart — APPROVED** [review]

**Comment by patricklx** [comment]
I think this is rare enough that we can just wait until we have the fix from glimmer vm

**Pull request closed without merging**